comp/dogstatsd/serverDebug/debug.go

```diff
@@ -250,3 +262,27 @@ func (d *serverDebug) disableMetricsStats() {
 
 	d.log.Info("Disabling DogStatsD debug metrics stats.")
 }
+
+// build a local dogstatsd logger and bubbling up any errors
+func getDogstatsdDebug() slog.LoggerInterface {
+
+	var dogstatsdLogger slog.LoggerInterface
+
+	// Configuring the log file path
+	logFile := config.Datadog.GetString("dogstatsd_log_file")
```

Comment:
This is something that changed recently, and is a bit of a gotcha with anything in `comp/`: We should no longer be using `config.Datadog` in components. 

instead, we need to inject the config component. You can mostly [copy what we did here](https://github.com/DataDog/datadog-agent/blob/50653843add85968d51b230a4928a0778a452e6e/comp/dogstatsd/server/server.go#L184-L193)
but the gist of it is: 

1. You need to update the `dependencies` [here](https://github.com/DataDog/datadog-agent/blob/main/comp/dogstatsd/serverDebug/debug.go#L29) to include the config component. 
2. [You need to inject it here ](https://github.com/DataDog/datadog-agent/blob/main/comp/dogstatsd/serverDebug/debug.go#L64)
3. Then you can reference the config component the same way you use the global.